File: Python/Algorithms/1441.py
class Solution:
    # 14.1 84.41
    def buildArray(self, target, n):
        ops = []
        app, ext = ops.append, ops.extend
        start = 0
        for i in target:
            diff = i - start - 1
            # Add any skipped over elements.
            if diff:
                ext(["Push", "Pop"] * diff)
            # Add the Push for i.
            app("Push")
            start = i
        return ops

File: Rust/Algorithms/1441.rs
impl Solution {
    pub fn build_array(target: Vec<i32>, n: i32) -> Vec<String> {
        let mut ops = Vec::new();
        let mut start = 0;
        for i in target {
            let diff = i - start - 1;
            if diff > 0 {
                for _i in 0..diff {
                    ops.push(String::from("Push"));
                    ops.push(String::from("Pop"));
                }
            }
            ops.push(String::from("Push"));
            start = i;
        }
        ops
    }
}
